webpack/webpack.common.js

```diff
@@ -1,81 +1,91 @@
+	entry: path.resolve(__dirname, '../src/index.tsx'),
 
-  output: {
-    path: path.resolve(__dirname,  '../dist'),
-    publicPath: "/"
-  },
+	output: {
+		path: path.resolve(__dirname,  '../dist'),
+		publicPath: "/"
+	},
 
-  module: {
-    rules: [
+	resolve: {
+		extensions: ['.ts', '.tsx', '.js']
+	},
+
+	module: {
+		rules: [
+			{
+				test: /\.tsx?$/,
+				loader: 'babel-loader'
+			},
 			{
 				test: /\.(js|jsx)$/,
 				exclude: /node_modules/,
 				use: {
 					loader: 'babel-loader',
 				}
 			},
-      {
-        test: /\.(eot|otf|ttf|woff|woff2)$/,
-        use: 'file-loader'
-      },
-      {
-        test: /\.svg$/,
-        use: [
-          {
-            loader: 'svg-url-loader',
-            options: {
-              // Inline files smaller than 10 kB
-              limit: 10 * 1024,
-              noquotes: true
-            },
-          },
-        ],
-      },
-      {
-        test: /\.(jpg|png|gif)$/,
-        use: [
-          {
-            loader: 'url-loader',
-            options: {
-              // Inline files smaller than 10 kB
-              limit: 10 * 1024
-            },
-          },
-          {
-            loader: 'image-webpack-loader',
-            options: {
-              mozjpeg: {
-                enabled: false
-                // NOTE: mozjpeg is disabled as it causes errors in some Linux environments
-                // Try enabling it in your environment by switching the config to:
-                // enabled: true,
-                // progressive: true,
-              },
-              gifsicle: {
-                interlaced: false
-              },
-              optipng: {
-                optimizationLevel: 7
-              },
-              pngquant: {
-                quality: '65-90',
-                speed: 4
-              }
-            }
-          }
-        ]
-      }
-    ]
-  },
+			{
+				test: /\.(eot|otf|ttf|woff|woff2)$/,
+				use: 'file-loader'
+			},
+			{
+				test: /\.svg$/,
+				use: [
+					{
+						loader: 'svg-url-loader',
+						options: {
+							// Inline files smaller than 10 kB
+							limit: 10 * 1024,
+							noquotes: true
+						},
+					},
+				],
+			},
+			{
+				test: /\.(jpg|png|gif)$/,
+				use: [
+					{
+						loader: 'url-loader',
+						options: {
+							// Inline files smaller than 10 kB
+							limit: 10 * 1024
+						},
+					},
+					{
+						loader: 'image-webpack-loader',
+						options: {
+							mozjpeg: {
+								enabled: false
+								// NOTE: mozjpeg is disabled as it causes errors in some Linux environments
+								// Try enabling it in your environment by switching the config to:
+								// enabled: true,
+								// progressive: true,
+							},
+							gifsicle: {
+								interlaced: false
+							},
+							optipng: {
+								optimizationLevel: 7
+							},
+							pngquant: {
+								quality: '65-90',
+								speed: 4
+							}
+						}
+					}
+				]
+			}
+		]
+	},
```

Comment:
This is freaking weird spacing issues, don't know if prettier could fix this but let's deal w it later